beamer/config/state.py

```diff
@@ -0,0 +1,135 @@
+import hashlib
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Generator
+
+import apischema
+from apischema import schema, validator
+from apischema.metadata import alias
+from apischema.objects import get_alias
+from eth_typing import ChecksumAddress
+from eth_utils import is_checksum_address
+
+from beamer.typing import BlockNumber, ChainId
+
+
+class ValidationError(Exception):
+    def __str__(self) -> str:
+        if self.__cause__ is None:
+            return super().__str__()
+        assert isinstance(self.__cause__, apischema.ValidationError)
+        return "\n".join(map(str, self.__cause__.errors))  # pylint: disable=no-member
+
+
+@dataclass(frozen=True)
+class TokenConfig:
+    transfer_limit: int = field(metadata=schema(min=0))
+    eth_in_token: int = field(metadata=schema(min=0))
+
+
+@dataclass(frozen=True)
+class ChainConfig:
+    finality_period: int = field(metadata=schema(min=1))
+    target_weight_ppm: int = field(metadata=schema(min=0, max=999_999))
+    transfer_cost: int = field(metadata=schema(min=0))
+
+
+@dataclass
+class FillManagerConfig:
+    whitelist: set[ChecksumAddress]
+
+
+@dataclass
+class RequestManagerConfig:
+    min_fee_ppm: int = field(metadata=schema(min=0, max=999_999))
```

Comment:
can be larger than 999999

Reply:
the reason for it is that is could be a more than 100% fee applied if transfer cost were very low.

Reply:
Should we have a "reasonable" upper bound here or shall I just remove the `max` completely?

Reply:
would just remove it completely, so that it basically aligns with the contracts